api/utils/studentCSVToJson.py

```diff
@@ -0,0 +1,130 @@
+                            except ValueError:
+                                non_int_values.add(str_value)
+                # Reconcile non int values
+                next_avail_int = 0
+                for val in non_int_values:
+                    while next_avail_int in values.values():
+                        next_avail_int += 1
+                    values[val] = next_avail_int
+                attributes.append((len(attributes), rows[0][i], values))
+
+        # Map student name to id
+        student_dict: Dict[str, int] = {
+            row[names_column]: i for i, row in enumerate(rows[1:])
+        }
+
+        students: List[Student] = []
+        for row in rows[1:]:
+            relationships = (
+                {
+                    student_dict[name]: Relationship.FRIEND
+                    for name in row[friends_column].split(",")
+                }
+                if friends_column and row[friends_column] != ""
+                else {}
+            )
+            relationships.update(
+                {
+                    student_dict[name]: Relationship.ENEMY
+                    for name in row[enemies_column].split(",")
+                }
+                if enemies_column and row[enemies_column] != ""
+                else {}
+            )
+
+            student_attributes = {}
+            for col in range(len(row)):
+                if col not in known_columns:
+                    attr = _get_attribute_by_name(rows[0][col], attributes)
+                    vals = [
+                        attr[2][value] for value in row[col].split(",") if value != ""
+                    ]
+                    if len(vals) > 0:
+                        student_attributes[attr[0]] = vals
+
+            students.append(
+                Student(
+                    _id=student_dict[row[names_column]],
+                    name=row[names_column],
+                    relationships=relationships,
+                    attributes=student_attributes,
+                )
+            )
+
+        # Convert to a dict, so we can fix the relationship values for the API
+        students_json = json.dumps(students, cls=StudentSerializer)
+        students_dict = json.loads(students_json)
+        for student in students_dict:
+            for related_student in student["relationships"].keys():
+                if (
+                    student["relationships"][related_student]
+                    == Relationship.FRIEND.value
+                ):
+                    student["relationships"][related_student] = "friend"
+                elif (
+                    student["relationships"][related_student]
+                    == Relationship.ENEMY.value
+                ):
+                    student["relationships"][related_student] = "enemy"
+                else:
+                    student["relationships"][related_student] = "default"
+
+        return json.dumps(students_dict, indent=4)
```

Comment:
I think this piece of code needs an explanation.

K's API requires that student relationships be in the format `relationships: { "sid" : ["friend" | "enemy" | "default"] }`. The `StudentSerializer` obviously doesn't output this format since Students map sids to Relationship objects.

This code first converts student objects to JSON, then converts to a python dict. Then, I change the Relationship value to the string the the API wants, then since it's a dict and not a Student list, I can convert back to JSON without using the StudentSerializer